src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,224 @@
+            return currentNode.item;
+        }
+        if (index == size - 1) {
+            currentNode.previous.next = setNode;
+            tail = setNode;
+            unlinkNode(currentNode);
+            return currentNode.item;
+        }
+        currentNode.previous.next = setNode;
+        currentNode.next.previous = setNode;
+        unlinkNode(currentNode);
+        return currentNode.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> currentNode = findNode(index);
+        if (index == 0) {
+            if (size == 1) {
+                size = 0;
+                head = tail = null;
+                return currentNode.item;
+            }
+            currentNode.next.previous = null;
+            head = currentNode.next;
+            unlinkNode(currentNode);
+            size--;
+            return currentNode.item;
+        }
+        if (index == size - 1) {
+            currentNode.previous.next = null;
+            tail = currentNode.previous;
+            unlinkNode(currentNode);
+            size--;
+            return currentNode.item;
+        }
+        currentNode.previous.next = currentNode.next;
+        currentNode.next.previous = currentNode.previous;
+        unlinkNode(currentNode);
+        size--;
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> searchNode = head;
+        while (searchNode != null) {
+            if (searchNode.item == object
+                    || (searchNode.item != null && searchNode.item.equals(object))) {
+                if (searchNode == head || (searchNode != null && searchNode.equals(head))) {
+                    if (searchNode.next == null) {
+                        head = tail = null;
+                        size--;
+                        return true;
+                    }
+                    searchNode.next.previous = null;
+                    head = searchNode.next;
+                    unlinkNode(searchNode);
+                    size--;
+                    return true;
+                }
+                if (searchNode == tail || (searchNode != null && searchNode.equals(tail))) {
+                    searchNode.previous.next = null;
+                    tail = searchNode.previous;
+                    unlinkNode(searchNode);
+                    size--;
+                    return true;
+                }
+                searchNode.previous.next = searchNode.next;
+                searchNode.next.previous = searchNode.previous;
+                unlinkNode(searchNode);
+                size--;
+                return true;
+            }
+            searchNode = searchNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void unlinkNode(Node node) {
```

Comment:
To my mind it's not enough just to unlink input node, we have to connect previous one with the next one

Reply:
agree as you suggestion implied more improvement of my code. thanks